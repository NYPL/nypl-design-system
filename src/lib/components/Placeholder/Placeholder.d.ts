import * as React from "react";
export interface PlaceholderProps {
    modifiers?: string[];
    blockName?: string;
}
/** Creates a gray box with an editable string for dropping in spaces
 * within components or on templates where a component will be but
 * that hasn't been created yet
 */
export default function Placeholder(
    props: React.PropsWithChildren<PlaceholderProps>
): JSX.Element;
