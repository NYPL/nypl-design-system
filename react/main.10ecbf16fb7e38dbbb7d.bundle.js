(window.webpackJsonp=window.webpackJsonp||[]).push([[0],{1292:function(n,o,c){"use strict";c.r(o);c(1293),c(1323),c(1335),c(1442),c(1485)},1448:function(n,o){},1449:function(n,o){},1450:function(n,o){},1451:function(n,o){},501:function(n,o,c){c(502),c(645),n.exports=c(1292)},564:function(n,o){}},[[501,1,2]]]);