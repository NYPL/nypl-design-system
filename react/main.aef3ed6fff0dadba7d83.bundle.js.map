{"version":3,"file":"main.aef3ed6fff0dadba7d83.bundle.js","sources":["webpack:///main.aef3ed6fff0dadba7d83.bundle.js"],"mappings":"AAAA","sourceRoot":""}