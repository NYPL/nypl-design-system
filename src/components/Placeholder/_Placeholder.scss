// Only used for the `Placeholder` component, not to style
// an `input`'s placeholder.
.placeholder {
  --color-place-holder: var(--nypl-colors-ui-gray-dark);

  margin-bottom: 0;
  padding: var(--nypl-space-s);
  align-items: center;
  border: 4px dashed var(--color-place-holder);
  color: var(--color-place-holder);
  display: flex;
  font-size: var(--nypl-fontSizes-desktop-heading-heading3);
  justify-content: center;
  line-height: 1.4;
  padding: 4rem 1rem;
  width: 100%;

  &--short {
    padding: 1rem 1rem;
  }
}
