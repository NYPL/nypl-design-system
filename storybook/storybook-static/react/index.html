<!doctype html><html lang="en"><head><meta charset="utf-8"><title>Storybook</title><meta name="viewport" content="width=device-width,initial-scale=1"><style>html, body {
    overflow: hidden;
    height: 100%;
    width: 100%;
    margin: 0;
    padding: 0;
  }</style><script>/* globals window */
  /* eslint-disable no-underscore-dangle */
  try {
    if (window.top !== window) {
      window.__REACT_DEVTOOLS_GLOBAL_HOOK__ = window.top.__REACT_DEVTOOLS_GLOBAL_HOOK__;
    }
  } catch (e) {
    // eslint-disable-next-line no-console
    console.warn('unable to connect to top frame for connecting dev tools');
  }</script><style>#root[hidden],
      #docs-root[hidden] {
        display: none !important;
      }</style></head><body><div id="root"></div><div id="docs-root"></div><script>window['DOCS_MODE'] = false;</script><script src="runtime~main.00bd9d033d7a0b972fc9.bundle.js"></script><script src="vendors~main.d2b6aafef1a259d165c0.bundle.js"></script><script src="main.3bf71672a6dd35499004.bundle.js"></script></body></html>