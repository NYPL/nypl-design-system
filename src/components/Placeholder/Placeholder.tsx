import * as React from "react";
import bem from "../../utils/bem";

export interface PlaceholderProps {
  /** BlockName for use with BEM. See how to work with blockNames and BEM here: http://getbem.com/introduction/ */
  blockName?: string;
  /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
  modifiers?: string[];
}

/** Creates a gray box with an editable string for dropping in spaces
 * within components or on templates where a component will be but
 * that hasn't been created yet
 */
export default function Placeholder(
  props: React.PropsWithChildren<PlaceholderProps>
) {
  const { modifiers = [], blockName } = props;
  return (
    <div className={bem("placeholder", modifiers, blockName)}>
      {props.children}
    </div>
  );
}
