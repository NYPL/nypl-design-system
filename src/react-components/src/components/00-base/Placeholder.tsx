
import * as React from "react";
import bem from "../../utils/bem";

export interface PlaceholderProps {
  modifiers?: string[];
  blockName?: string;
}

export default function Placeholder(props: React.PropsWithChildren<PlaceholderProps>) {
  const { modifiers = [], blockName } = props;
  return (
    <div className={bem("placeholder")}>{props.children}</div>
  );
}
