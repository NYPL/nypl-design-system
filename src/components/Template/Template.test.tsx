import * as React from "react";
import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import renderer from "react-test-renderer";

import {
  Template,
  TemplateAboveHeader,
  TemplateHeader,
  TemplateBreakout,
  TemplateContent,
  TemplateContentTop,
  TemplateContentPrimary,
  TemplateContentSidebar,
  TemplateFooter,
  TemplateAppContainer,
} from "./Template";
import Placeholder from "../Placeholder/Placeholder";

const aboveHeader = <Placeholder variant="short">Above Header</Placeholder>;
const header = <Placeholder variant="short">NYPL Header</Placeholder>;
const breakout = (
  <>
    <Placeholder variant="short">Breadcrumbs</Placeholder>
    <Placeholder variant="short">Hero</Placeholder>
  </>
);
const sidebar = "left";
const contentTop = <Placeholder>Content Top</Placeholder>;
const contentSidebar = <Placeholder>Left Sidebar</Placeholder>;
const contentPrimary = (
  <>
    <Placeholder>Main Content</Placeholder>
    <Placeholder variant="short">More Content</Placeholder>
  </>
);
const footer = <Placeholder variant="short">Footer</Placeholder>;

describe("TemplateAppContainer accessibility", () => {
  it("passes axe accessibility test", async () => {
    const { container } = render(
      <TemplateAppContainer
        aboveHeader={aboveHeader}
        header={header}
        breakout={breakout}
        sidebar={sidebar}
        contentTop={contentTop}
        contentSidebar={contentSidebar}
        contentPrimary={contentPrimary}
        footer={footer}
      />
    );
    expect(await axe(container)).toHaveNoViolations();
  });
});

describe("Template components accessibility", () => {
  it("passes axe accessibility test", async () => {
    const { container } = render(
      <Template>
        <TemplateAboveHeader>{aboveHeader}</TemplateAboveHeader>
        <TemplateHeader>
          {header}
          <TemplateBreakout>{breakout}</TemplateBreakout>
        </TemplateHeader>
        <TemplateContent sidebar={sidebar}>
          <TemplateContentTop>{contentTop}</TemplateContentTop>
          <TemplateContentSidebar>{contentSidebar}</TemplateContentSidebar>
          <TemplateContentPrimary>{contentPrimary}</TemplateContentPrimary>
        </TemplateContent>
        <TemplateFooter>{footer}</TemplateFooter>
      </Template>
    );
    expect(await axe(container)).toHaveNoViolations();
  });
});

describe("TemplateAppContainer component", () => {
  it("renders each section", () => {
    render(
      <TemplateAppContainer
        aboveHeader={aboveHeader}
        header={header}
        breakout={breakout}
        sidebar={sidebar}
        contentTop={contentTop}
        contentSidebar={contentSidebar}
        contentPrimary={contentPrimary}
        footer={footer}
      />
    );
    expect(screen.getByText("Above Header")).toBeInTheDocument();
    expect(screen.getByText("NYPL Header")).toBeInTheDocument();
    expect(screen.getByText("Breadcrumbs")).toBeInTheDocument();
    expect(screen.getByText("Hero")).toBeInTheDocument();
    expect(screen.getByText("Content Top")).toBeInTheDocument();
    expect(screen.getByText("Left Sidebar")).toBeInTheDocument();
    expect(screen.getByText("Main Content")).toBeInTheDocument();
    expect(screen.getByText("More Content")).toBeInTheDocument();
    expect(screen.getByText("Footer")).toBeInTheDocument();
  });

  it("renders only one header in a custom header component", () => {
    const customHeader = <header>Custom header</header>;
    render(
      <TemplateAppContainer
        header={customHeader}
        renderHeaderElement={false}
        breakout={breakout}
        sidebar={sidebar}
        contentTop={contentTop}
        contentSidebar={contentSidebar}
        contentPrimary={contentPrimary}
        footer={footer}
      />
    );

    // The `<header>` HTML element has the same meaning as `role="banner"`.
    expect(screen.getAllByRole("banner")).toHaveLength(1);
  });

  it("consoles a warning when a header element was passed without setting `renderHeaderElement` to false", () => {
    const warn = jest.spyOn(console, "warn");
    const customHeader = <header>Custom header</header>;
    render(
      <TemplateAppContainer
        header={customHeader}
        breakout={breakout}
        sidebar={sidebar}
        contentTop={contentTop}
        contentSidebar={contentSidebar}
        contentPrimary={contentPrimary}
        footer={footer}
      />
    );
    expect(warn).toHaveBeenCalledWith(
      "`TemplateHeader`: An HTML `header` element was passed in. Set " +
        "`renderHeaderElement` to `false` to avoid nested HTML `header` elements."
    );
  });

  it("renders only one footer in a custom footer component", () => {
    const customFooter = <footer>Custom Footer</footer>;
    render(
      <TemplateAppContainer
        header={header}
        breakout={breakout}
        sidebar={sidebar}
        contentTop={contentTop}
        contentSidebar={contentSidebar}
        contentPrimary={contentPrimary}
        footer={customFooter}
        renderFooterElement={false}
      />
    );

    expect(screen.getAllByRole("contentinfo")).toHaveLength(1);
  });

  it("consoles a warning when a footer element was passed without setting `renderFooterElement` to false", () => {
    const warn = jest.spyOn(console, "warn");
    const customFooter = <footer>Custom Footer</footer>;
    render(
      <TemplateAppContainer
        header={header}
        breakout={breakout}
        sidebar={sidebar}
        contentTop={contentTop}
        contentSidebar={contentSidebar}
        contentPrimary={contentPrimary}
        footer={customFooter}
      />
    );
    expect(warn).toHaveBeenCalledWith(
      "`TemplateFooter`: An HTML `footer` element was passed in. Set " +
        "`renderFooterElement` to `false` to avoid nested HTML `footer` elements."
    );
  });
});

describe("Template components", () => {
  it("renders each section", () => {
    render(
      <Template>
        <TemplateAboveHeader>{aboveHeader}</TemplateAboveHeader>
        <TemplateHeader>
          {header}
          <TemplateBreakout>{breakout}</TemplateBreakout>
        </TemplateHeader>
        <TemplateContent sidebar={sidebar}>
          <TemplateContentTop>{contentTop}</TemplateContentTop>
          <TemplateContentSidebar>{contentSidebar}</TemplateContentSidebar>
          <TemplateContentPrimary>{contentPrimary}</TemplateContentPrimary>
        </TemplateContent>
        <TemplateFooter>{footer}</TemplateFooter>
      </Template>
    );

    expect(screen.getByText("Above Header")).toBeInTheDocument();
    expect(screen.getByText("NYPL Header")).toBeInTheDocument();
    expect(screen.getByText("Breadcrumbs")).toBeInTheDocument();
    expect(screen.getByText("Hero")).toBeInTheDocument();
    expect(screen.getByText("Content Top")).toBeInTheDocument();
    expect(screen.getByText("Left Sidebar")).toBeInTheDocument();
    expect(screen.getByText("Main Content")).toBeInTheDocument();
    expect(screen.getByText("More Content")).toBeInTheDocument();
    expect(screen.getByText("Footer")).toBeInTheDocument();
  });

  it("Renders the UI snapshot correctly", () => {
    const templateComponents = renderer
      .create(
        <Template>
          <TemplateAboveHeader>{aboveHeader}</TemplateAboveHeader>
          <TemplateHeader>
            {header}
            <TemplateBreakout>{breakout}</TemplateBreakout>
          </TemplateHeader>
          <TemplateContent sidebar={sidebar}>
            <TemplateContentTop>{contentTop}</TemplateContentTop>
            <TemplateContentSidebar>{contentSidebar}</TemplateContentSidebar>
            <TemplateContentPrimary>{contentPrimary}</TemplateContentPrimary>
          </TemplateContent>
          <TemplateFooter>{footer}</TemplateFooter>
        </Template>
      )
      .toJSON();
    const singleComponent = renderer
      .create(
        <TemplateAppContainer
          aboveHeader={aboveHeader}
          header={header}
          breakout={breakout}
          sidebar={sidebar}
          contentTop={contentTop}
          contentSidebar={contentSidebar}
          contentPrimary={contentPrimary}
          footer={footer}
        />
      )
      .toJSON();
    const withChakraProps = renderer
      .create(
        <TemplateAppContainer
          aboveHeader={aboveHeader}
          header={header}
          breakout={breakout}
          sidebar={sidebar}
          contentTop={contentTop}
          contentSidebar={contentSidebar}
          contentPrimary={contentPrimary}
          footer={footer}
          p="20px"
          color="ui.error.primary"
        />
      )
      .toJSON();
    const withOtherProps = renderer
      .create(
        <TemplateAppContainer
          aboveHeader={aboveHeader}
          header={header}
          breakout={breakout}
          sidebar={sidebar}
          contentTop={contentTop}
          contentSidebar={contentSidebar}
          contentPrimary={contentPrimary}
          footer={footer}
          data-testid="props"
        />
      )
      .toJSON();

    expect(templateComponents).toMatchSnapshot();
    expect(singleComponent).toMatchSnapshot();
    expect(withChakraProps).toMatchSnapshot();
    expect(withOtherProps).toMatchSnapshot();
  });
});
