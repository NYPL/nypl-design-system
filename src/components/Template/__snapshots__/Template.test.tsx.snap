// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template components renders the UI snapshot correctly 1`] = `
<div
  className="css-1xdhyk6"
>
  <div
    className="css-0"
  >
    <div
      className="placeholder placeholder--short"
    >
      Above Header
    </div>
  </div>
  <header
    className="css-0"
  >
    <div
      className="placeholder placeholder--short"
    >
      NYPL Header
    </div>
    <div
      className="css-0"
    >
      <div
        className="placeholder placeholder--short"
      >
        Breadcrumbs
      </div>
      <div
        className="placeholder placeholder--short"
      >
        Hero
      </div>
    </div>
  </header>
  <main
    className="css-0"
    id="mainContent"
  >
    <div
      className="css-0"
    >
      <div
        className="placeholder placeholder--undefined"
      >
        Content Top
      </div>
    </div>
    <div
      className="css-0"
    >
      <div
        className="placeholder placeholder--undefined"
      >
        Left Sidebar
      </div>
    </div>
    <div
      className="css-0"
    >
      <div
        className="placeholder placeholder--undefined"
      >
        Main Content
      </div>
      <div
        className="placeholder placeholder--short"
      >
        More Content
      </div>
    </div>
  </main>
  <footer
    className="css-0"
  >
    <div
      className="placeholder placeholder--short"
    >
      Footer
    </div>
  </footer>
</div>
`;

exports[`Template components renders the UI snapshot correctly 2`] = `
<div
  className="css-1u8qly9"
>
  <div
    className="css-0"
  >
    <div
      className="css-0"
    >
      <div
        className="placeholder placeholder--short"
      >
        Above Header
      </div>
    </div>
    <header
      className="css-0"
    >
      <div
        className="placeholder placeholder--short"
      >
        NYPL Header
      </div>
      <div
        className="css-0"
      >
        <div
          className="placeholder placeholder--short"
        >
          Breadcrumbs
        </div>
        <div
          className="placeholder placeholder--short"
        >
          Hero
        </div>
      </div>
    </header>
  </div>
  <main
    className="css-0"
    id="mainContent"
  >
    <div
      className="css-0"
    >
      <div
        className="placeholder placeholder--undefined"
      >
        Content Top
      </div>
    </div>
    <div
      className="css-0"
    >
      <div
        className="placeholder placeholder--undefined"
      >
        Left Sidebar
      </div>
    </div>
    <div
      className="css-0"
    >
      <div
        className="placeholder placeholder--undefined"
      >
        Main Content
      </div>
      <div
        className="placeholder placeholder--short"
      >
        More Content
      </div>
    </div>
  </main>
  <footer
    className="css-0"
  >
    <div
      className="placeholder placeholder--short"
    >
      Footer
    </div>
  </footer>
</div>
`;

exports[`Template components renders the UI snapshot correctly 3`] = `
<div
  className="css-1y4kn3e"
>
  <div
    className="css-0"
  >
    <div
      className="css-0"
    >
      <div
        className="placeholder placeholder--short"
      >
        Above Header
      </div>
    </div>
    <header
      className="css-0"
    >
      <div
        className="placeholder placeholder--short"
      >
        NYPL Header
      </div>
      <div
        className="css-0"
      >
        <div
          className="placeholder placeholder--short"
        >
          Breadcrumbs
        </div>
        <div
          className="placeholder placeholder--short"
        >
          Hero
        </div>
      </div>
    </header>
  </div>
  <main
    className="css-0"
    id="mainContent"
  >
    <div
      className="css-0"
    >
      <div
        className="placeholder placeholder--undefined"
      >
        Content Top
      </div>
    </div>
    <div
      className="css-0"
    >
      <div
        className="placeholder placeholder--undefined"
      >
        Left Sidebar
      </div>
    </div>
    <div
      className="css-0"
    >
      <div
        className="placeholder placeholder--undefined"
      >
        Main Content
      </div>
      <div
        className="placeholder placeholder--short"
      >
        More Content
      </div>
    </div>
  </main>
  <footer
    className="css-0"
  >
    <div
      className="placeholder placeholder--short"
    >
      Footer
    </div>
  </footer>
</div>
`;

exports[`Template components renders the UI snapshot correctly 4`] = `
<div
  className="css-1u8qly9"
  data-testid="props"
>
  <div
    className="css-0"
  >
    <div
      className="css-0"
    >
      <div
        className="placeholder placeholder--short"
      >
        Above Header
      </div>
    </div>
    <header
      className="css-0"
    >
      <div
        className="placeholder placeholder--short"
      >
        NYPL Header
      </div>
      <div
        className="css-0"
      >
        <div
          className="placeholder placeholder--short"
        >
          Breadcrumbs
        </div>
        <div
          className="placeholder placeholder--short"
        >
          Hero
        </div>
      </div>
    </header>
  </div>
  <main
    className="css-0"
    id="mainContent"
  >
    <div
      className="css-0"
    >
      <div
        className="placeholder placeholder--undefined"
      >
        Content Top
      </div>
    </div>
    <div
      className="css-0"
    >
      <div
        className="placeholder placeholder--undefined"
      >
        Left Sidebar
      </div>
    </div>
    <div
      className="css-0"
    >
      <div
        className="placeholder placeholder--undefined"
      >
        Main Content
      </div>
      <div
        className="placeholder placeholder--short"
      >
        More Content
      </div>
    </div>
  </main>
  <footer
    className="css-0"
  >
    <div
      className="placeholder placeholder--short"
    >
      Footer
    </div>
  </footer>
</div>
`;
